From 84c7032bd1a744fc1c363d9836151b497064e402 Mon Sep 17 00:00:00 2001
From: Eric Garver <eric@garver.life>
Date: Tue, 28 Oct 2025 15:59:26 -0400
Subject: [PATCH 32/37] v2.4.0: chore(icmp): add all icmptypes to ICMP_TYPES
 dict

We can then use this as a source of truth for types/codes.

(cherry picked from commit b121faab666faddf7181d21691ee6e8a13b30dbe)
---
 src/firewall/core/icmp.py | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/src/firewall/core/icmp.py b/src/firewall/core/icmp.py
index ed030026a111..87f87ab1c49e 100644
--- a/src/firewall/core/icmp.py
+++ b/src/firewall/core/icmp.py
@@ -25,32 +25,42 @@ __all__ = [ "ICMP_TYPES", "ICMPV6_TYPES",
 ICMP_TYPES = {
      "echo-reply": "0/0",
      "pong": "0/0",
+     "destination-unreachable": "3/0",
      "network-unreachable": "3/0",
+     "tos-network-unreachable": "3/0",
      "host-unreachable": "3/1",
+     "tos-host-unreachable": "3/1",
      "protocol-unreachable": "3/2",
      "port-unreachable": "3/3",
      "fragmentation-needed": "3/4",
      "source-route-failed": "3/5",
+     # RFC-1112 Section 3.2.2.1 defines type 3, code 6-12
      "network-unknown": "3/6",
      "host-unknown": "3/7",
      "network-prohibited": "3/9",
      "host-prohibited": "3/10",
      "TOS-network-unreachable": "3/11",
      "TOS-host-unreachable": "3/12",
+     # RFC-1812 Section 5.2.7.1 defines type 3, code 13-15
      "communication-prohibited": "3/13",
      "host-precedence-violation": "3/14",
      "precedence-cutoff": "3/15",
      "source-quench": "4/0",
      "network-redirect": "5/0",
+     "redirect": "5/0",
      "host-redirect": "5/1",
+     "tos-host-redirect": "5/1",
      "TOS-network-redirect": "5/2",
+     "tos-network-redirect": "5/2",
      "TOS-host-redirect": "5/3",
      "echo-request": "8/0",
      "ping": "8/0",
      "router-advertisement": "9/0",
      "router-solicitation": "10/0",
+     "time-exceeded": "11/0",
      "ttl-zero-during-transit": "11/0",
      "ttl-zero-during-reassembly": "11/1",
+     "parameter-problem": "12/0",
      "ip-header-bad": "12/0",
      "required-option-missing": "12/1",
      "timestamp-request": "13/0",
@@ -60,13 +70,19 @@ ICMP_TYPES = {
 }
 
 ICMPV6_TYPES = {
+    "destination-unreachable": "1/0",
     "no-route": "1/0",
     "communication-prohibited": "1/1",
+    "beyond-scope": "1/2",
     "address-unreachable": "1/3",
     "port-unreachable": "1/4",
+    "failed-policy": "1/5",
+    "reject-route": "1/6",
     "packet-too-big": "2/0",
+    "time-exceeded": "3/0",
     "ttl-zero-during-transit": "3/0",
     "ttl-zero-during-reassembly": "3/1",
+    "parameter-problem": "4/0",
     "bad-header": "4/0",
     "unknown-header-type": "4/1",
     "unknown-option": "4/2",
@@ -81,6 +97,12 @@ ICMPV6_TYPES = {
     "neighbour-advertisement": "136/0",
     "neigbour-advertisement": "136/0",
     "redirect": "137/0",
+    # MLD is RFC-2710
+    "mld-listener-query": "130/0",
+    "mld-listener-report": "131/0",
+    "mld-listener-done": "132/0",
+    # MLDv2 is RFC-9777
+    "mld2-listener-report": "143/0",
 }
 
 def check_icmp_name(_name):
-- 
2.52.0

